### PR TITLE
[x-history] Prints CONTROL version history of a port 👻

### PR DESCRIPTION
**Pull request by vicroms**
This PR adds a new command `vcpkg x-history <port>` that shows the full history of CONTROL versions of a port.

Example:

```
PS D:\vcpkg> ./vcpkg x-history sqlite3
             version          date    vcpkg commit
            3.29.0-1    2019-07-21    2abd03cce852f00a302c4de000da3ef0eb7a4768
            3.28.0-2    2019-07-15    e1870a91ff26c099ca92599fa7d7bb06b81fbc58
            3.28.0-1    2019-06-20    dfd5bb46a149bbc2d9b4f4fa6a5fd4df5992cbdb
              3.28.0    2019-06-17    8de7aa163317885a84e39a3dc81726e74537ba32
              3.27.2    2019-06-15    18b029a5e3997fa4fdc7d3d06d56568a1d6f74ad
              3.26.0    2018-12-18    afdca9165c739760d58e29b4a25ef10d470a1aaf
              3.25.2    2018-11-06    a0d6a2925c5eb2cb7b3c793f74eff3220f893007
            3.24.0-2    2018-10-22    473d63c4edf1ab704b597abc4987372712b005f6
            3.24.0-1    2018-06-16    3726ce95576204cc8a2c37d07a1ae74b0b70c1d2
              3.24.0    2018-06-05    b2c295ed4e212ee14358e441df2193508a5f40cf
            3.23.1-1    2018-04-24    90db2c72173c7398122e693d797f82736dabd109
              3.23.1    2018-04-24    cde623c774618d8cebed56e08dd0c0106ce0a951
              3.23.0    2018-04-03    182450ab6434cd2b790780e45f6a6cd060874761
            3.21.0-1    2018-03-19    62a015b4f3754d92381a02b5b11641a1ddcc256d
              3.21.0    2017-10-25    ae48463360d4394a9c55e91f46f6f103dd046ab3
              3.20.1    2017-09-11    601661ec323808afad4c3588a5a2473eb86a0eb4
            3.19.1-2    2017-09-09    26516fe485b0e9048dd4809256a7e4526957c6e9
            3.19.1-1    2017-05-24    14c3c51a009e1764b7f1141cdc024493bced5d8d
            3.18.0-1    2017-04-01    e5a4f033170d66ec53f23a5366a3c8e896fdc1ea
              3.17.0    2017-04-01    13940af99a201e29599911632894f62dea0e5db2
              3.15.0    2016-12-06    c128d007df40ee0f4a55f954cb153ddd983a4d36
    3.15.0 (3150000)    2016-10-17    56f8c6a462b234def8eda554c4af88aaa0c0f992
             3120200    2016-10-06    b1fe038a839d54fa7da9cadf8eb5e487bf30cb01
```

Feedback is welcome :) 

P.S.: Is the name `port-history` good?

### CODE REVIEW

**Comment by JackBoosY** [comment]
I think `history` is more succinct, unless this command has other uses :)

**Comment by Voskrese** [comment]
@JackBoosY 

`git clone https://github.com/microsoft/vcpkg`

cd vcpkg

`cat ports/sqlite3/CONTROL`

> Source: sqlite3
> Version: 3.29.0-1

`git checkout 796e65ea73e883e275a7908c4214913d32d5327a`

`cat ports\sqlite3\CONTROL`

> Source: sqlite3
> Version: 3.15.0

` --overlay-ports=<path>`

**Comment by JackBoosY** [comment]
@Voskrese No, I don't mean this, I just think the command `port-history` is too long, because the command is followed by the port name.

**Comment by ras0219-msft** [comment]
(should first deploy as an `x-` command)

File: toolsrc/src/vcpkg/help.cpp [hunk 1]
```diff
@@ -91,6 +91,7 @@ namespace vcpkg::Help
                        "  vcpkg list                      List installed packages\n"
                        "  vcpkg update                    Display list of packages for updating\n"
                        "  vcpkg upgrade                   Rebuild all outdated packages\n"
+                       "  vcpkg history <pkg>             Shows the history of CONTROL versions of a package\n"
```
Comment (myd7349) [inline toolsrc/src/vcpkg/help.cpp]: Hi! @vicroms It seems that `history` here should also be renamed to `x-history`?

Comment (myd7349) [inline toolsrc/src/vcpkg/help.cpp]: Hi! @vicroms It seems that `history` here should also be changed to `x-history`?

**Comment by JackBoosY** [comment]
@myd7349 Thanks for pointing this out. Fixing in #8101.

**Pull request merged**